internal/match/json.go

```diff
@@ -204,3 +225,29 @@ func JSONMapEach(wantKey string, fn func(k, v gjson.Result) error) JSON {
 		return err
 	}
 }
+
+// AnyOf takes 1 or more `checkers`, and builds a new checker which accepts a given
+// json body iff it's accepted by at least one of the original `checkers`.
+func AnyOf(checkers ...JSON) JSON {
```

Comment:
Oh this is a great idea! 👍🏼

Reply:
> Tests are failing however:
> 
> ```
> === RUN   TestRoomSpecificUsernameHandlingOverFederation
>     user_directory_federated_display_names_test.go:22: Deploy times: 7.681441ms blueprints, 3.556640339s containers
>     client.go:407: POST hs1/_matrix/client/r0/register => 200 OK (40.50622ms)
>     client.go:407: PUT hs2/profile/@charlie:hs2/displayname => 404 Not Found (2.345413ms)
>     user_directory_federated_display_names_test.go:33: CSAPI.MustDoFunc response return non-2xx code: 404 Not Found - body: 
>         <html>
>           <head><title>404 - No Such Resource</title></head>
>           <body>
>             <h1>No Such Resource</h1>
>             <p>No such child resource.</p>
>           </body>
>         </html>
> ```

Thanks for spotting. I'm missing a ` /_matrix/client/r0/`.